HLTrigger/special/plugins/HLTPPSPerPotTrackFilter.cc

```diff
@@ -0,0 +1,192 @@
+
+  struct PerPotFilter {
+    int minTracks;
+    int maxTracks;
+  };
+  std::unordered_map<uint32_t, PerPotFilter> pixel_filter_;
+  std::unordered_map<uint32_t, PerPotFilter> strip_filter_;
+  std::unordered_map<uint32_t, PerPotFilter> diam_filter_;
+
+  // Helper tool to count valid tracks
+  static constexpr auto valid_trks_ = [](const auto& trk) { return trk.isValid(); };
+};
+
+void HLTPPSPerPotTrackFilter::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+  edm::ParameterSetDescription desc;
+
+  desc.add<edm::InputTag>("pixelLocalTrackInputTag", edm::InputTag("ctppsPixelLocalTracks"))
+      ->setComment("input tag of the pixel local track collection");
+  desc.add<edm::InputTag>("stripLocalTrackInputTag", edm::InputTag("totemRPLocalTrackFitter"))
+      ->setComment("input tag of the strip local track collection");
+  desc.add<edm::InputTag>("diamondLocalTrackInputTag", edm::InputTag("ctppsDiamondLocalTracks"))
+      ->setComment("input tag of the diamond local track collection");
+
+  edm::ParameterSetDescription filterValid;
+  filterValid.add<unsigned int>("detid", 0)->setComment("station/pot raw DetId");
+  filterValid.add<int>("minTracks", -1)->setComment("minimum number of tracks in pot");
+  filterValid.add<int>("maxTracks", -1)->setComment("maximum number of tracks in pot");
+
+  std::vector<edm::ParameterSet> vPixelDefault;
+  auto& near_pix45 = vPixelDefault.emplace_back();
+  near_pix45.addParameter<unsigned int>("detid", CTPPSPixelDetId(0, 2, 2));  // arm-station-rp
+  near_pix45.addParameter<int>("minTracks", 2);
+  near_pix45.addParameter<int>("maxTracks", -1);
+  auto& far_pix45 = vPixelDefault.emplace_back();
+  far_pix45.addParameter<unsigned int>("detid", CTPPSPixelDetId(0, 2, 3));  // arm-station-rp
+  far_pix45.addParameter<int>("minTracks", 2);
+  far_pix45.addParameter<int>("maxTracks", -1);
+  auto& near_pix56 = vPixelDefault.emplace_back();
+  near_pix56.addParameter<unsigned int>("detid", CTPPSPixelDetId(1, 2, 2));  // arm-station-rp
+  near_pix56.addParameter<int>("minTracks", 2);
+  near_pix56.addParameter<int>("maxTracks", -1);
+  auto& far_pix56 = vPixelDefault.emplace_back();
+  far_pix56.addParameter<unsigned int>("detid", CTPPSPixelDetId(1, 2, 3));  // arm-station-rp
+  far_pix56.addParameter<int>("minTracks", 2);
+  far_pix56.addParameter<int>("maxTracks", -1);
+  desc.addVPSet("pixelFilter", filterValid, vPixelDefault);
+
+  std::vector<edm::ParameterSet> vStripDefault;
+  desc.addVPSet("stripFilter", filterValid, vStripDefault);
+
+  std::vector<edm::ParameterSet> vDiamDefault;
+  desc.addVPSet("diamondFilter", filterValid, vDiamDefault);
+
+  desc.add<int>("triggerType", trigger::TriggerTrack);
+
+  descriptions.add("hltPPSPerPotTrackFilter", desc);
+}
+
+HLTPPSPerPotTrackFilter::HLTPPSPerPotTrackFilter(const edm::ParameterSet& iConfig) {
+  // First define pixels (2017+) selection
+  const auto pixelVPset = iConfig.getParameter<std::vector<edm::ParameterSet>>("pixelFilter");
```

Comment:
```suggestion
  auto const& pixelVPset = iConfig.getParameter<std::vector<edm::ParameterSet>>("pixelFilter");
```
by taking a reference you avoid making a temporary copy of the object.